labels.go

```diff
@@ -57,11 +57,60 @@ func setLoadbalancerSticky(lb *dynamic.ServersLoadBalancer, val string) error {
 	}
 
 	if v {
-		lb.Sticky = &dynamic.Sticky{}
+		if lb.Sticky == nil {
+			lb.Sticky = &dynamic.Sticky{ Cookie: &dynamic.Cookie{} }
+		}
+	}
+	return nil
+}
+
+func setLoadbalancerStickySecure(lb *dynamic.ServersLoadBalancer, val string) error {
+	v, err := strconv.ParseBool(val)
+	if err != nil {
+		v = false
+	}
+
+	if v {
+		if lb.Sticky == nil {
+			lb.Sticky = &dynamic.Sticky{ Cookie: &dynamic.Cookie{} }
+		}
+		lb.Sticky.Cookie.Secure = v
+	}
+	return nil
+}
+
+func setLoadbalancerStickyHttpOnly(lb *dynamic.ServersLoadBalancer, val string) error {
+	v, err := strconv.ParseBool(val)
+	if err != nil {
+		v = false
+	}
+
+	if v {
+		if lb.Sticky == nil {
+			lb.Sticky = &dynamic.Sticky{ Cookie: &dynamic.Cookie{} }
+		}
+		lb.Sticky.Cookie.HTTPOnly = v
+	}
+	return nil
+}
+
+func setLoadbalancerStickySameSite(lb *dynamic.ServersLoadBalancer, val string) error {
+	if lb.Sticky == nil {
+		lb.Sticky = &dynamic.Sticky{ Cookie: &dynamic.Cookie{} }
+	}
+	lb.Sticky.Cookie.SameSite = val
```

Comment:
@nokjuh should we be defensive here and validate the string to ensure it's one of: 'none', 'lax', or 'strict'?

You'll need to **import "strings"** for the following suggestion to work.

```suggestion
	valid := map[string]bool{"none": true, "lax": true, "strict": true}

       if valid[strings.ToLower(val)] {
	     lb.Sticky.Cookie.SameSite = val
	} else {
	     log.Printf("Unrecognised value '%s' provided for Cookie.SameSite",  val)
             return nil
	}

```

Reply:
This is now done.